src/XMakeBuildEngine/UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -3294,20 +3290,33 @@ public void Medley()
                 errorTests.Add("$([System.IO.Path]::Combine(`|`,`b`))");
             }
 
+#if FEATURE_WIN32_REGISTRY
+            if (NativeMethodsShared.IsWindows)
+            {
+                errorTests.Add("$(Registry:X)");
+            }
+            else
+            {
+                // If no registry, and not running on windows, this gets expanded to the empty string
+                // example: xplat build running on OSX
+                validTests.Add(new string[]{"$(Registry:X)", ""});
```

Comment:
Hm the existing code tested this case only for !netcore. But this should probably be tested for netcore too. It seems to resolve to `""` even for netcore, so, I will update this.